tests/python/pants_test/task/test_task.py

```diff
@@ -7,12 +7,17 @@
 
 import os
 
+from pants.backend.jvm.subsystems.zinc import Zinc
```

Comment:
This test in package `pants_test.task` (core-y) really shouldn't depend on `pants.backend.jvm.*` (leaf-y).  It's probably best for sanity sake to create a FakeTask and FakeSubsystem in this file and use them (TaskTestBase.context accepts a `for_subsystems`). Ditto `pants.source.source_root`.

Reply:
This makes perfect sense, I was just having trouble getting anything to work the way I wanted and was grasping at straws. I've removed the leaf-y imports and refactored the way I was synthesizing and instantiating classes based on your comments, thanks.